cdb.createVis returns a vis model that triggers load event when vizjson has been loaded

alonsogarciapablo opened the pull request:
Prep work for https://github.com/CartoDB/cartodb/issues/7574.

Changes:
- `cdb.createVis` returns a vis model object that responds to `.done` and `.error`. I didn't change how this works and done callback will be called when the map is first instantiated.
- `vis` object triggers `.load` event when vizjson has been loaded so that apps that use `createVis` with `skipMapInstantiation` option can instantiate the map when thy need to when everything is loaded (eg: deep-insights.js -> see https://github.com/CartoDB/deep-insights.js/pull/380).
- To make ^^ happen, I moved model related stuff from `vis-view.js` to `vis.js` (everything related with loading the vizjson).

@viddo: could you please take a look? Thx!

viddo commented on src/vis/vis-view.js:
```diff
@@ -49,113 +30,17 @@ var Vis = View.extend({
-  done: function (fn) {
-    return this.bind('done', fn);
-  },
+    this.model.on('load', this.render, this);
```
Can this be triggered multiple times? Otherwise a `.once` might be preferable.

alonsogarciapablo replied:
👍

viddo commented on src/vis/vis-view.js:
```diff
@@ -49,113 +30,17 @@ var Vis = View.extend({
-      }.bind(this));
-    }
+  render: function () {
+    // TODO: Get this from this.model
```
What's "this"? I'd recommend to create an issue to keep track of work your plan to do in the future instead of leaving a TODO here

alonsogarciapablo replied:
I forgot it. I'll take care of it in this PR. Thanks!

viddo commented:
> Prep work for CartoDB/cartodb#7574.

The desc sounds reasonable, but I lack some info what this PR is preparing _for_? :)

The changes looks legit. 👍

alonsogarciapablo commented:
> I lack some info what this PR is preparing for? :)

Basically, before this PR, createVis was returning an instance of `vis-view.js` which had a lot of stuff that belongs to a model. Since we want to expose some error through the `vis` object I feel this was a great opportunity to tackle this and I think it'll make it easier now to work with that `vis` object internally :)

alonsogarciapablo commented:
@viddo I addressed your feedback in https://github.com/CartoDB/cartodb.js/pull/1313/commits/9bc7d6e28461297c9c8fe8f1a68b400d5cb68d14. Would love a second pair of 👀  on that commit to make sure it makes sense. Thanks!

viddo commented:
Thx for clarification, that makes it easier to understand the _why_ behind the changes. :)
Regarding the last changes I don't really have anything particular to say, looks legit and accurate (from the little knowledge I have of these parts).

alonsogarciapablo commented:
Retest this please
